v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -0,0 +1,78 @@
+import importlib
+from enum import Enum
+
+import hail as hl
+
+from v03_pipeline.lib.model import AccessControl, DatasetType, Env, ReferenceGenome
+
+DATASET_TYPES = 'dataset_types'
+VERSION = 'version'
+RAW_DATASET_PATH = 'raw_dataset_path'
+ENUM_SELECT = 'enum_select'
+
+
+class ReferenceDataset(str, Enum):
+    cadd = 'cadd'
+    hgmd = 'hgmd'
+
+    @classmethod
+    def for_reference_genome_dataset_type(
+        cls,
+        reference_genome: ReferenceGenome,
+        dataset_type: DatasetType,
+    ) -> list['ReferenceDataset']:
+        reference_datasets = [
+            dataset
+            for dataset, config in CONFIG.items()
+            if dataset_type in config.get(reference_genome, {}).get(DATASET_TYPES, [])
+        ]
+        if not Env.ACCESS_PRIVATE_REFERENCE_DATASETS:
+            return [
+                dataset
+                for dataset in reference_datasets
+                if dataset.access_control == AccessControl.PUBLIC
+            ]
+        return reference_datasets
+
+    @property
+    def access_control(self) -> AccessControl:
+        if self == ReferenceDataset.hgmd:
+            return AccessControl.PRIVATE
+        return AccessControl.PUBLIC
+
+    def version(self, reference_genome: ReferenceGenome) -> str:
+        return CONFIG[self][reference_genome][VERSION]
+
+    def raw_dataset_path(self, reference_genome: ReferenceGenome) -> str | list[str]:
+        return CONFIG[self][reference_genome][RAW_DATASET_PATH]
+
+    def get_ht(
+        self,
+        raw_dataset_path: str | list[str],
```

Comment:
Why not call `self.raw_dataset_path` inside the function?

Reply:
so true

Reply:
ah, cause `self.raw_dataset_path` isn't a property and requires an arg.  I fixed it on my branch :/